### PR TITLE
Add doc that covers how to add an app from Git

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:
The noteworthy thing in this PR is what it treats the way to add an app from Git as the recommended way.

**Which issue(s) this PR fixes**:

Fixes https://github.com/pipe-cd/pipe/issues/2834

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nakabonne** [comment]
While I'm in the middle of making Environment optional (https://github.com/pipe-cd/pipe/issues/2915), this PR is open for review because I'm going to make a change a bit after addressing.

**Comment by nakabonne** [comment]
I made revisions to make env optional.

**Comment by khanhtc1202** [comment]
@nakabonne Nice work 👍  cc @nghialv 
Just want to confirm one thing: After we release this feature, new users who come for the examples in `pipe-cd/examples` repository may use old application configuration files from there and those files do not contain `name` or `envName` field; thus they can't be used to add to PipeCD via `Git add application` feature. Should we update those configuration files? I'm okay to update those files with a separated PR.

**Comment by nakabonne** [comment]
hey @khanhtc1202, good point. I'm going to update all of them when working on https://github.com/pipe-cd/pipe/issues/2936.

File: docs/content/en/docs-dev/user-guide/configuration-reference.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ spec:
 
 | Field | Type | Description | Required |
 |-|-|-|-|
+| name | string | The application name. | Yes if you set the application through the application configuration file |
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/configuration-reference.md:22]:
```suggestion
| name | string | The application name. | Yes (if you want to create PipeCD application through the application configuration file) |
```
nits, and same for the others.

**Comment by nakabonne** [comment]
Because of https://github.com/pipe-cd/pipe/pull/2950, I changed the default file name.

File: docs/content/en/docs-dev/user-guide/configuration-reference.md (hunk 1 above)
Comment (nakabonne) [reply docs/content/en/docs-dev/user-guide/configuration-reference.md:22]: Looks good. But currently others are in the form of this. So let me work on it as a separate PR.
image

Comment (nakabonne) [reply docs/content/en/docs-dev/user-guide/configuration-reference.md:22]: Okay, for the sentence, let me borrow yours 👍

Comment (khanhtc1202) [reply docs/content/en/docs-dev/user-guide/configuration-reference.md]: Thanks for clarifying 👍

File: docs/content/en/docs-dev/operator-manual/piped/configuration-reference.md [hunk 2]
```diff
@@ -24,8 +24,9 @@ spec:
 | pipedKeyFile | string | The path to the file containing the generated key string for this piped. | Yes |
 | pipedKeyData | string | Base64 encoded string of Piped key. Either pipedKeyFile or pipedKeyData must be set. | Yes |
 | apiAddress | string | The address used to connect to the control-plane's API. | Yes |
-| syncInterval | duration | How often to check whether an application should be synced. Default is `1m`. | No | <!-- TODO: Add doc for appConfigSyncInterval -->
-| git | [Git](/docs/operator-manual/piped/configuration-reference/#git) | Git configuration needed for Git commands.  | No |
+| syncInterval | duration | How often to check whether an application should be synced. Default is `1m`. | No |
+| appConfigSyncInterval | duration | How often to check whether application configuration files should be synced. Default is `5m`. | No |
+| git | [Git](/docs/operator-manual/piped/configuration-reference/#git) | Git configuration needed for Git commands. | No |
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/operator-manual/piped/configuration-reference.md:29]:
```suggestion
| git | [Git](#git) | Git configuration needed for Git commands. | No |
```
nits, should refer to the tag of the same page, not jump to other version docs

Comment (nakabonne) [reply docs/content/en/docs-dev/operator-manual/piped/configuration-reference.md:29]: Good catch!

File: docs/content/en/docs-dev/user-guide/adding-an-application.md [hunk 3]
```diff
@@ -61,6 +87,17 @@ spec:
 - `kind` is the application kind. As explained before, supporting kinds of application are: `Kubernetes`, `Terrform`, `CloudRun`, `Lambda` and `ECS`.
 - `spec` is the specific configuration for each application kind.
 
-Please refer [pipecd/examples](https://pipecd.dev/docs/user-guide/examples/) for the deployments being supported.
+Please refer [pipecd/examples](/docs/user-guide/examples/) for the deployments being supported.
 
 The [next section](/docs/user-guide/configuring-deployment/) guides you how to configure the deployment for each specific application kinds.
+
+## Updating an application
+Regardless of which method you use to register an application, you need to update the application via the application configuration file (that's why registering from Git is recommended).
```
Comment (nghialv) [inline docs/content/en/docs-dev/user-guide/adding-an-application.md:95]:
```suggestion
Regardless of which method you used to register the application, the web console can only be used to disable/enable/delete the application, besides the adding operation. All updates on application information must be done via the application configuration file stored in Git.
```

File: docs/content/en/docs-dev/user-guide/adding-an-application.md [hunk 4]
```diff
@@ -61,6 +87,17 @@ spec:
...
+Refer to [the supported field](/user-guide/configuration-reference/) and make revisions where you need to update.
```
Comment (nghialv) [inline docs/content/en/docs-dev/user-guide/adding-an-application.md:96]:
```suggestion
```

File: docs/content/en/docs-dev/user-guide/adding-an-application.md [hunk 5]
```diff
@@ -61,6 +87,17 @@ spec:
...
+
+```yaml
+apiVersion: pipecd.dev/v1beta1
+kind: KubernetesApp
+spec:
+  name: new-name
+```
```
Comment (nghialv) [inline docs/content/en/docs-dev/user-guide/adding-an-application.md:103]:
```suggestion
Refer to [configuration reference](/user-guide/configuration-reference/) to see the full list of configurable fields.
```

File: docs/content/en/docs-dev/user-guide/adding-an-application.md [hunk 6]
```diff
@@ -7,26 +7,52 @@ description: >
...
+It represents the service which you are going to deploy. With PipeCD, all application's manifests and its application configuration (`app.pipecd.yaml`) must be committed into a directory of a Git repository. That directory is called as application directory.
 
-Before deploying an application, the application must be registered from the web UI and a application configuration file (`.pipe.yaml`) must be committed to the application directory.
-An application must belong to exactly one environment and can be handled by one of the registered `piped`s. Currently, PipeCD is supporting the following kinds of application:
+Before deploying an application, the application must be registered from either the web UI or an application configuration file defined in your Git repository.
+Registering application helps PipeCD know the basic information about that application, where the application configuration is placing, what `piped` should handle it as well as what cloud the application should be deployed to.
+
+An application can be handled by one of the registered `piped`s. Currently, PipeCD is supporting the following kinds of application:
```
Comment (nghialv) [inline docs/content/en/docs-dev/user-guide/adding-an-application.md:15]: I guess we have to update the Japanese doc too.

File: docs/content/en/docs-dev/user-guide/adding-an-application.md [hunk 7]
```diff
@@ -7,26 +7,52 @@ description: >
 ---
 
 An application is a collect of resources and configurations that are managed together.
-It represents the service which you are going to deploy. With PipeCD, all application's manifests and its application configuration (`.pipe.yaml`) must be committed into a directory of a Git repository. That directory is called as application directory.
+It represents the service which you are going to deploy. With PipeCD, all application's manifests and its application configuration (`app.pipecd.yaml`) must be committed into a directory of a Git repository. That directory is called as application directory.
 
-Before deploying an application, the application must be registered from the web UI and a application configuration file (`.pipe.yaml`) must be committed to the application directory.
-An application must belong to exactly one environment and can be handled by one of the registered `piped`s. Currently, PipeCD is supporting the following kinds of application:
+Before deploying an application, the application must be registered from either the web UI or an application configuration file defined in your Git repository.
```
Comment (nghialv) [inline docs/content/en/docs-dev/user-guide/adding-an-application.md:12]: I think both "manually form" and "git form" are registering via "web console". So maybe we don't have to explicitly distinguish them here.

```suggestion
Before deploying an application, it must be registered via the web console.
```

File: docs/content/en/docs-dev/user-guide/adding-an-application.md [hunk 8]
```diff
@@ -7,26 +7,52 @@ description: >
...
+Registering application helps PipeCD know the basic information about that application, where the application configuration is placing, what `piped` should handle it as well as what cloud the application should be deployed to.
```
Comment (nghialv) [inline docs/content/en/docs-dev/user-guide/adding-an-application.md:13]:
```suggestion
Registering application helps PipeCD know where the application configuration is placed, which `piped` should handle it as well as which cloud the application should be deployed to.
```

File: docs/content/en/docs-dev/user-guide/adding-an-application.md (hunk 6 above)
Comment (nghialv) [inline docs/content/en/docs-dev/user-guide/adding-an-application.md:15]:
```suggestion
Each application can be handled by one and only one `piped`. Currently, PipeCD is supporting the following application kinds:
```

File: docs/content/en/docs-dev/user-guide/adding-an-application.md [hunk 9]
```diff
@@ -7,26 +7,52 @@ description: >
...
 - Kubernetes application
 - Terraform application
 - CloudRun application
 - Lambda application
 - ECS application
 
-## Registering a new application from Web UI
+There are two ways to register an application:
+- From the application configuration in your Git repository (recommended)
+- From the web UI
```
Comment (nghialv) [inline docs/content/en/docs-dev/user-guide/adding-an-application.md:25]: I want to say that both of them are "From the web UI", so I guess we need a better way to express them.
Below is just my suggestion.
We can find a better one.

```suggestion
There are two ways to register an application:
- Scanning the unused application configuration files in Git to add (recommended)
- Manually configure application information
```

File: docs/content/en/docs-dev/user-guide/adding-an-application.md (hunk 6 above)
Comment (nakabonne) [reply docs/content/en/docs-dev/user-guide/adding-an-application.md]: Yeah, let me update them later 👍

**Comment by nakabonne** [comment]
PTAL!

**Comment by nghialv** [comment]
Nice.
/lgtm

File: docs/content/en/docs-dev/user-guide/adding-an-application.md [hunk 10]
```diff
@@ -7,26 +7,52 @@ description: >
...
+The file name must be suffixed by `.pipecd.yaml` because Piped periodically checks for files with this suffix.
 
+```yaml
+apiVersion: pipecd.dev/v1beta1
+kind: KubernetesApp
+spec:
+  name: foo
+```
+
+Visit [here](/user-guide/configuration-reference/) for supported fields.
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/adding-an-application.md:40]:
```suggestion
Visit [here](/docs/user-guide/configuration-reference/) for supported fields.
```
nits

File: docs/content/en/docs-dev/user-guide/adding-an-application.md [hunk 11]
```diff
@@ -61,6 +87,18 @@ spec:
...
+Regardless of which method you used to register the application, the web console can only be used to disable/enable/delete the application, besides the adding operation. All updates on application information must be done via the application configuration file stored in Git.
+
+```yaml
+apiVersion: pipecd.dev/v1beta1
+kind: KubernetesApp
+spec:
+  name: new-name
+```
+
+Refer to [configuration reference](/user-guide/configuration-reference/) to see the full list of configurable fields.
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/adding-an-application.md:104]:
```suggestion
Refer to [configuration reference](/docs/user-guide/configuration-reference/) to see the full list of configurable fields.
```
nits

File: docs/content/en/docs-dev/user-guide/adding-an-application.md (hunk 10 above)
Comment (nakabonne) [reply docs/content/en/docs-dev/user-guide/adding-an-application.md:40]: nice catch!

**Comment by nakabonne** [comment]
@khanhtc1202 Fixed

**Comment by khanhtc1202** [comment]
Nice
/approve

**Pull request merged**